lib/resty/kafka/producer.lua

```diff
@@ -23,6 +23,19 @@ local crc32 = ngx.crc32_short
 local pcall = pcall
 local pairs = pairs
 
+local lrucache = require "resty.lrucache"
+local cache = lrucache.new(1)
+if not cache then
+    return ngx_log(ERR, "failed to create the cache: " .. (err or "unknown"))
+else
+    cache:set("max_timers_count", 0)
+end
+
+-- according to the current implementation, each "running timer" will take one (fake) connection record 
+-- from the global connection record list configured by the standard worker_connections directive in nginx.conf. 
+-- so limit the timers. global max timer count for all workers
```

Comment:
Here actually limit the total timers including both pending and running timers.
>> I mean we should not limit the pending timer, which may cause data lose.

Fail to turn pending timer running causes an error log which is annoying
>> this usually happen when the running timer num reached the `lua_max_running_timer`, but we only have two running timer at most in this lib.